validator/lib/validateB3dm.js

```diff
@@ -29,27 +38,90 @@ function validateB3dm(content) {
         return {
             result : false,
-            message: 'Tile has an invalid version'
+            message: 'b3dm tile has an invalid version'
         };
     }
 
     if (byteLength !== content.length) {
         return {
             result : false,
-            message: 'Tile has the wrong byteLength'
+            message: 'b3dm tile has the wrong byteLength'
         };
     }
 
+    var batchTableJSONByteLength = content.readUInt32LE(12);
+    var batchTable;
+    if(batchTableJSONByteLength > 0) {
+        batchTable = extractBatchTable(content);
+    }
+
+    if((defined(batchTable)) && (defined(batchTable.batchTableJSON))) {
+        var validBatchTable = validateBatchTable(batchTableSchema, batchTable.batchTableJSON, batchTable.batchTableBinary);
+        if(!validBatchTable.validation) {
+            return {
+                result : false,
+                message: validBatchTable.message
+            };
+        }
+    }
+
     return {
         result : true,
-        message: 'Tile is a valid b3dm tile'
+        message: 'b3dm tile is a valid b3dm tile'
     };
+}
 
+/**
+ * Extracts the batch table from the provided tile
+ *
+ * @param {Buffer} tile - A buffer containing the contents of the tile
+ * @returns {Object}  - A JSON object representing the batch table, containing (1) the batch table JSON,
+ *                     (2)batch table binary, (3) message for error or success
+ */
+
+function extractBatchTable(tile) {
+    var byteLength = tile.length;
+    var batchTableJSONByteOffset = 12;
+    var batchTableOffset = 24;
+
+    var batchTableJSONByteLength = tile.readUInt32LE(batchTableJSONByteOffset);
+    var batchTableBinaryByteLength = tile.readUInt32LE(batchTableJSONByteOffset + 4);
+    var message = '';
+    var batchTableJSON, batchTableBinary;
+
+    if(batchTableJSONByteLength > 0) {
+        if((batchTableOffset + batchTableJSONByteLength) > byteLength) {
+            message += 'batchTableJSONByteLength is out of bounds at ' + batchTableOffset + batchTableJSONByteLength;
+        } else {
+            var batchTableJSONHeader = tile.slice(batchTableOffset, batchTableOffset + batchTableJSONByteLength);
+            batchTableJSON = JSON.parse(batchTableJSONHeader.toString());
+        }
+
+        batchTableOffset += batchTableJSONByteLength;
+
+        if((batchTableOffset + batchTableBinaryByteLength) > byteLength) {
+            message += '\nbatchTableBinaryByteLength is out of bounds at ' + batchTableOffset + batchTableBinaryByteLength;
+        } else {
+            batchTableBinary = tile.slice(batchTableOffset, batchTableOffset + batchTableBinaryByteLength);
+        }
+    } else {
+        message = 'Error: trying to extract batch table with length <= 0';
+    }
+
+    if(message === '') {
+        message = 'successfully extracted batch table'
+    }
+
+    return {
+        batchTableJSON: batchTableJSON,
+        batchTableBinary: batchTableBinary,
+        message: message
+    };
```

Comment:
This section may be able to be condensed by extracting the sections, and then verifying they are correct.

For example - https://github.com/AnalyticalGraphicsInc/3d-tiles-tools/blob/master/tools/lib/extractB3dm.js#L44-L46. You could do something like that and then verify the buffers are the correct size. Ideally the batch table extraction/validation could even go inline with the rest of the b3dm checking.